codice/ddf
# DDF-4675 - Add type extend support definitions

#### What does this PR do?
Adds the ability to compose other metacard types into ones created in the definitions files.  You will now be able to add the `extendsTypes` attribute to metacard types definition files with values for predefined metacard types (ones that already exist in the system or ones defined already in the file) (eg, ["core", "security"]
#### Who is reviewing it? 
@tbatie @glenhein @ethantmanns @bellcc @jordanwilking
#### Select relevant component teams: 
@codice/data 

#### How should this be tested?
Create a json definitions file (Examples and documentation were updated, though i would recommend trying something new and custom since the documentation/unit tests already work) that utilizes the new `extendsTypes` field for metacard type definitions. Ensure that metacard types without the field (older files) still function as expected.

#### What are the relevant tickets?
Fixes: #4675

#### Checklist:
  - No threat vectors changed
  - This is a standalone module that does not have any system integration effects, and thus no integration tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
